base/auspice_export.py

```diff
@@ -39,6 +39,23 @@ def process_freqs(freq):
     else:
         process.log.notify("Cannot export frequencies - pivots do not exist")
 
+def export_tip_frequency_json(process, prefix, indent):
+    def process_freqs(freq): # 6dp here, 4 above
+        return [round(x,6) for x in freq]
+
+    if not (hasattr(process, 'pivots') and hasattr(process, 'tree_frequencies')):
+        process.log.notify("Cannot export tip frequencies - pivots and/or tree_frequencies do not exist")
+        return;
+
+    freq_json = {'pivots':process_freqs(process.pivots)}
+
+    clade_tip_map = {n.clade: n.name for n in process.tree.tree.get_terminals()}
+
+    for clade, freq in process.tree_frequencies["global"].iteritems():
+        if clade in clade_tip_map:
+            freq_json[clade_tip_map[clade]] = process_freqs(freq)
```

Comment:
Is there a reason to switch the format of the JSON to indexing frequencies by clade name instead of clade id? Along those same lines, can maintain information about which region the frequencies are from in the data themselves?

It would be great if the tip frequencies JSON was backwards compatible with the complete JSON, so code that refers to one could refer to the other without any changes. Annotating the region will make the data more interpretable to anyone who is just looking at the JSON and it will also allow us to export tip frequencies for multiple regions in the future (which is a likely goal, I think).

Reply:
038748012656420d39d932e5e4926cdb0b6db6b9